hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -777,20 +779,35 @@ Joiner createJoiner() {
                 logger.info("Creating TcpIpJoiner");
                 return new TcpIpJoiner(this);
             } else if (join.getAwsConfig().isEnabled()) {
-                Class clazz;
-                try {
-                    logger.info("Creating AWSJoiner");
-                    clazz = Class.forName("com.hazelcast.cluster.impl.TcpIpJoinerOverAWS");
-                    Constructor constructor = clazz.getConstructor(Node.class);
-                    return (Joiner) constructor.newInstance(this);
-                } catch (Exception e) {
-                    throw rethrow(e);
-                }
+                logger.info("Creating AWSJoiner");
+                return createAwsJoiner();
             }
         }
         return null;
     }
 
+    private Joiner createAwsJoiner() {
+        try {
+            Class clazz = Class.forName("com.hazelcast.cluster.impl.TcpIpJoinerOverAWS");
+            Constructor constructor = clazz.getConstructor(Node.class);
+            return (Joiner) constructor.newInstance(this);
+        } catch (ClassNotFoundException e) {
+            StringBuilder message = new StringBuilder("Your Hazelcast network configuration has AWS discovery ")
+                    .append("enabled, but there is no Hazelcast AWS module on a classpath. ")
+                    .append(LINE_SEPARATOR)
+                    .append("Hint: If you are using Maven then add this dependency into your pom.xml:")
+                    .append(LINE_SEPARATOR)
+                    .append("<dependency>").append(LINE_SEPARATOR)
+                    .append("    <groupId>com.hazelcast</groupId>").append(LINE_SEPARATOR)
+                    .append("    <artifactId>hazelcast-aws</artifactId>").append(LINE_SEPARATOR)
+                    .append("    <version>${hazelcast-aws.version}</version>").append(LINE_SEPARATOR)
```

Comment:
If someone would copy paste this logic and hasn't defined the hazelcast-aws.version property, it will still fail. 

1: Can't we look up the version of the module? I guess the module doesn't need to match the HZ version?

2: If we can't do a lookup, perhaps better to indicate <version>insert-right-version-here</version> to make it more obvious that some work by the end user is required here.